internal/pkg/agent/cmd/watch.go

```diff
@@ -135,13 +151,27 @@ func watchCmd(log *logp.Logger, cfg *configuration.Configuration) error {
 	// Why is this being skipped on Windows? The comment above is not clear.
 	// issue: https://github.com/elastic/elastic-agent/issues/3027
 	removeMarker := !isWindows()
-	err = upgrade.Cleanup(log, paths.Top(), marker.VersionedHome, marker.Hash, removeMarker, false)
+	err = installModifier.Cleanup(log, topDir, marker.VersionedHome, marker.Hash, removeMarker, false)
 	if err != nil {
 		log.Error("cleanup after successful watch failed", err)
 	}
 	return err
 }
 
+func isTerminalState(marker *upgrade.UpdateMarker) bool {
```

Comment:
before reading the body of this func I was under the impression that `isTerminalState` would return `true` if the marker is not found which then I realised I was wrong 😄 could you please add a godoc to describe what a terminalstate is and what it produces?